diff --git a/libuuu/CMakeLists.txt b/libuuu/CMakeLists.txt
index ef548fb..ddd4779 100644
--- a/libuuu/CMakeLists.txt
+++ b/libuuu/CMakeLists.txt
@@ -24,23 +24,14 @@ set(SOURCES
 	usbhotplug.cpp
 	version.cpp
 	sdp.cpp
-	gitversion.h
+	gen/gitversion.h
 	fastboot.cpp
 	zip.cpp
 	fat.cpp
 	rominfo.cpp
 )
 
-set(generated_files_dir "${CMAKE_BINARY_DIR}/libuuu/gen")
-set(gitversion_h "${generated_files_dir}/gitversion.h")
-
-add_custom_command(
-	OUTPUT gitversion.h
-	PRE_BUILD
-	COMMAND mkdir -p ${generated_files_dir}
-	COMMAND sh -c 'cd ${CMAKE_CURRENT_SOURCE_DIR} && rm -f ${gitversion_h} && ./gen_ver.sh "${gitversion_h}.tmp" && mv -f "${gitversion_h}.tmp" "${gitversion_h}"'
-
-)
+set(generated_files_dir "${CMAKE_SOURCE_DIR}/libuuu/gen")
 include_directories(${generated_files_dir})
 
 #add_library( uuc SHARED ${SOURCES} ))
